ddtrace/internal/tracemethods.py

```diff
@@ -56,34 +95,38 @@ def _parse_trace_methods(raw_dd_trace_methods):
                     )
                 )
             dd_trace_methods.append("%s.%s" % (qualified_method_prefix, method))
-
     return dd_trace_methods
 
 
-def _install_trace_methods(raw_dd_trace_methods):
-    # type: (str) -> None
+def _install_trace_methods(raw_dd_trace_methods: str) -> None:
     """Install tracing on the given methods."""
-    for qualified_method in _parse_trace_methods(raw_dd_trace_methods):
-        # We don't know if the method is a class method or a module method, so we need to assume it's a module
-        # and if the import fails then go a level up and try again.
-        base_module_guess = ".".join(qualified_method.split(".")[:-1])
-        method_name = qualified_method.split(".")[-1]
-        module = None
-
-        while base_module_guess:
-            try:
-                module = __import__(base_module_guess)
-            except ImportError:
-                # Add the class to the method name
-                method_name = "%s.%s" % (base_module_guess.split(".")[-1], method_name)
-                base_module_guess = ".".join(base_module_guess.split(".")[:-1])
-            else:
-                break
-
-        if module is None:
-            raise ImportError("Could not import module for %r" % qualified_method)
-
-        trace_method(base_module_guess, method_name)
+    if "[" in raw_dd_trace_methods:
+        # Using legacy syntax
+        for qualified_method in _parse_legacy_trace_methods(raw_dd_trace_methods):
+            # We don't know if the method is a class method or a module method, so we need to assume it's a module
+            # and if the import fails then go a level up and try again.
+            base_module_guess = ".".join(qualified_method.split(".")[:-1])
+            method_name = qualified_method.split(".")[-1]
+            module = None
+
+            while base_module_guess:
+                try:
+                    module = __import__(base_module_guess)
```

Comment:
If this import is likely triggered at bootstrap time, then perhaps the better thing to do would be to register an after import hook, like we do with integrations. The problem of force-loading a module at boot is, besides the potential side-effects, the incompatibility with the module cloning mechanism for supporting things such as `gevent`.

Reply:
We do register an after import hook, but this import happens first to figure out whether the method is a module or class method. The goal for this PR is to deprecate the current syntax of DD_TRACE_METHODS so that this distinction is clear from the the env var itself, and we don't need to try these imports to figure it out during bootstrapping. That way we can just parse the string and make the after import hooks immediately.